src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -42,7 +43,14 @@ public KafkaCallback(InflightMessageList inflightMessages, BinlogPosition positi
 	@Override
 	public void onCompletion(RecordMetadata md, Exception e) {
 		if ( e != null ) {
-			e.printStackTrace();
+			if ( e instanceof RecordTooLargeException ) {
+				LOGGER.error("RecordTooLargeException @ " + position + " -- " + key);
+				LOGGER.error("Maxwell dropped a row because it was too large (" + e.getLocalizedMessage() + ")");
+				LOGGER.error("Considering raising max.request.size broker-side.");
+				inflightMessages.completeMessage(position);
```

Comment:
Would there be a point in logging the failed binlog positions to another Kafka topic in order to be able to go in and retry them later? Or would that be overkill?